src/HasteModuleLoader/HasteModuleLoader.js

```diff
@@ -226,9 +226,12 @@ Loader.prototype._execModule = function(moduleObj) {
     || (onlyCollectFrom && onlyCollectFrom[modulePath] === true);
 
   if (shouldCollectCoverage) {
+    if (!this._CoverageCollector) {
+      this._CoverageCollector = require(this._config.coverageCollector);
+    }
```

Comment:
``require`` in constructor will call ``requireModuleOrMock``, I guess it is because ``new Loader()`` is in the context with mock ``require``. Actually I don't quite understand why is that, maybe this makes sense to you.